src/controllers/staff.controller.ts

```
@@ -1,55 +1,189 @@
+    //checks if mongoose threw and will rethrow with appropriate status code and message
+    checkMongooseErrors(err);
+
+    throw new HttpError(
+      HttpStatus.INTERNAL_SERVER_ERROR,
+      "Update staff failed",
+      {
+        cause: err,
+      },
+    );
+  }
 };
 
 export const getAllStaff = async () => {
-  const allStaff = await StaffModel.find({});
-  if (!allStaff) throw new Error("No staff found");
-  return allStaff;
+  try {
+    const allStaff = await StaffModel.find({});
+    return allStaff;
+  } catch (err: unknown) {
+    //rethrow any errors as HttpErrors
+    if (err instanceof HttpError) {
+      throw err;
+    }
+    //checks if mongoose threw and will rethrow with appropriate status code and message
+    checkMongooseErrors(err);
+
+    throw new HttpError(
+      HttpStatus.INTERNAL_SERVER_ERROR,
+      "Staff retrieval failed",
+      { cause: err },
+    );
+  }
 };
 
-export const getStaffByEmail = async (email: string) => {
-  const staff = await StaffModel.findOne({ email });
-  if (!staff) throw new Error("Staff not found");
-  return staff;
+export const getStaffByEmail = async (emailValue: string) => {
+  try {
+    const staff = await StaffModel.findOne({ email: emailValue });
+    return staff;
+  } catch (err: unknown) {
+    //rethrow any errors as HttpErrors
+    if (err instanceof HttpError) {
+      throw err;
+    }
+    //checks if mongoose threw and will rethrow with appropriate status code and message
+    checkMongooseErrors(err);
+
+    throw new HttpError(
+      HttpStatus.INTERNAL_SERVER_ERROR,
+      "Staff retrieval failed",
+      { cause: err },
+    );
+  }
 };
 
-export const getStaffByID = async (firebaseUID: string) => {
-  const staff = await StaffModel.findOne({ firebaseUID });
-  if (!staff) throw new Error("Staff not found");
-  return staff;
+export const getStaffByID = async (firebaseUIDValue: string) => {
+  try {
+    const staff = await StaffModel.findOne({ firebaseUID: firebaseUIDValue });
```

This should throw an error if there is no staff with the given firebaseID (in other words an error should be thrown if staff is null).